src/inc/quic_platform_winkernel.h

```diff
@@ -476,56 +477,122 @@ _CxPlatEventWaitWithTimeout(
 // Event Queue Interfaces
 //
 
-typedef KEVENT CXPLAT_EVENTQ; // Event queue
-typedef void* CXPLAT_CQE;
+typedef struct CXPLAT_EVENTQ {
+    CXPLAT_LOCK Lock;
+    LIST_ENTRY Events;
+    CXPLAT_EVENT EventsAvailable;
+} CXPLAT_EVENTQ;
 
+typedef struct CXPLAT_CQE {
+    void* UserData;
+} CXPLAT_CQE;
+
+#define CXPLAT_SQE CXPLAT_SQE
+#define CXPLAT_SQE_DEFAULT {0}
+typedef struct CXPLAT_SQE {
+    LIST_ENTRY Link;
+    void* UserData;
+    BOOLEAN IsQueued; // Prevent double queueing.
```

Comment:
Instead of a whole extra variable, can we use `Link->Flink` being `NULL` or not?

Reply:
If you have the appropriate synchronization, and you set Flink to NULL whenever it's not in the list, then sure. It's also not unusual to set an empty (non-inserted) link to be an empty list, i.e., Flink points to itself.

Reply:
Agreed. IsQueued was originally copied from the winuser CXPLAT_SQE